### DISCUSSION TITLE
context deadline exceeded error when connecting to cluster mode disabled cache

### DISCUSSION THREAD

**Discussion by lchalabi** [redis/rueidis#226] [Q&A]
Hi, I'm having trouble connecting to a redis cache using `rueidis.NewClient(...)`.   The cache in question is running on redis 5.0.6 and does not have cluster mode nor authentication enabled. 

My initialization looks like the following:
```go
	rCache, err := rueidis.NewClient(rueidis.ClientOption{
		InitAddress: []string{host},
		TLSConfig: &tls.Config{},
		DisableCache: true,
	})
	if err != nil {
		return nil, fmt.Errorf("connecting to redis: %s", err)
	}
```

I am able to connect to the cache using both `redis-cli` and another redis client library.  Does `rueidis` support redis caches without authentication?  I have been able to use it for clustered caches that have authentication.

**Comment by rueian** [comment]
Yes, it should support. Does your server require TLS?
